client/src/partials/templates/bhDateEditor.tmpl.html

```diff
@@ -0,0 +1,37 @@
+<!-- Date editor component view implementation-->
+<div 
+	class="form-group"	
+	ng-class="{'has-error' : $ctrl.validationTrigger && $ctrl.form.dateEditorValue.$invalid}" >
+
+	<!-- Input component implementation-->
+	<div class="input-group">
+	    <input
+	    	data-date-editor-input
+	    	name="dateEditorValue"
+	    	type="text"
+	    	class="form-control"	    	
+	    	uib-datepicker-popup="{{$ctrl.dateFormat}}"
+	    	ng-model="$ctrl.dateValue"
+	    	is-open="$ctrl.editMode"
+	    	min-date="$ctrl.minDate"
+	    	max-date="$ctrl.maxDate"
+	    	close-text="Close"
+	    	ng-disabled="!$ctrl.editMode"
+	    	show-button-bar="true"
+	    	required
+	    	/>
+	    <span class="input-group-btn">
+	    	<button id="set-date-btn" class="btn btn-default" type="button" ng-click="$ctrl.changeMode()" ng-disabled="$ctrl.editMode">
+	    		{{"DATEFINDER.SET" | translate}}
+	    	</button>
+	    </span>
+    </div> 
+
+    <!-- Error message implementation-->
+    <p class="help-block" ng-messages="$ctrl.form.dateEditorValue.$error" ng-show="$ctrl.validationTrigger">
```

Comment:
At Kinshasa, you wanted us to not create additional branches because developers would have too much freedom.  Now, you do not want to use `ng-messages` because doesn't allow freedom.

[Here was @sfount's comment on this method of form validation](https://github.com/IMA-WorldHealth/bhima-2.X/pull/89#discussion_r52283104).  You are right, it was not discussed.  It was simply put as a pull request that anyone could comment on, and well documented.

I don't know what the trap of modularity is.

Anyhow, it was just  a suggestion, it will not block the PR.